CDCgov/NEDSS-Modernization
# [CNFT1-1549] Adds legal name to search results

## Description

Changes the Patient Search API to return the most effective legal name of a patient as of the day the Search takes place.   When a Patient has multiple legal names effective for the same day, the last entered takes precedence (names are sequenced) .  

## Tickets

* [CNFT1-1549](https://cdc-nbs.atlassian.net/browse/CNFT1-1549)

## Steps to verify

1. Search for a Patient
2. Click the name of the Patient to enter the Patient Profile
3. Navigate to the Demographics tab
4. Add a new legal name
5. Return to search and search for the Patient
6. Ensure the new legal name is present

Patient Profile Card
![image](https://github.com/CDCgov/NEDSS-Modernization/assets/124325935/d6dde7f0-fdad-45c2-a5e2-6e35d82bcf30)

Patient Search Result
![image](https://github.com/CDCgov/NEDSS-Modernization/assets/124325935/b05e7fab-7d8a-46a8-b5b7-53e2dbabeca2)


[CNFT1-1549]: https://cdc-nbs.atlassian.net/browse/CNFT1-1549?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
